en/ros/ros2_multi_vehicle.md

```diff
@@ -0,0 +1,54 @@
+# Multi-Vehicle Simulation with ROS 2
+
+[XRCE-DDS](../middleware/xrce_dds.md) allows for multiple clients to be connected to the same agent over UDP.
+This is particular useful in simulation as only one agent needs to be started.
+
+## Setup and requirements
+
+The only requirements are
+
+- To be able to run [multi-vehicle simulation](../simulation/multi-vehicle-simulation.md) without ROS 2 with the desired simulator ([Gazebo](../sim_gazebo_gz/multi_vehicle_simulation.md), [Gazebo Classic](../sim_gazebo_classic/multi_vehicle_simulation_gazebo.md#multiple-vehicle-with-gazebo-classic-no-ros), [FlightGear](../simulation/multi_vehicle_flightgear.md) and [JMAVSim](../simulation/multi_vehicle_jmavsim.md)).
+- To be able to use [ROS 2](./ros2_comm.md) in a single vehicle simulation.
+
+## Principle of operation
+
+In simulation each PX4 instance receives an unique `px4_instance` number starting from `0`.
+This value is used to assign a unique value to [XRCE_DDS_KEY](../advanced_config/parameter_reference.md#XRCE_DDS_KEY):
+
+```sh
+param set XRCE_DDS_KEY $((px4_instance+1))
+```
+
+:::note
+By doing so, `XRCE_DDS_KEY` will always coincide with [MAV_COMP_ID](../advanced_config/parameter_reference.md#MAV_COMP_ID).
+:::
+
+Moreover, when `px4_instance` is greater than zero, a unique ROS 2 [namespace prefix](../middleware/xrce_dds.md#customizing-the-topic-namespace) in the form `px4_$px4_instance` is added:
+
+```sh
+microdds_ns="-n px4_$px4_instance"
+```
+
+:::note
+The environment variable `PX4_MICRODDS_NS`, if set, will override the above mentioned namespace behavior.
+:::
+
+The first instance (`px4_instance=0`) does not have an additional namespace in order to be consistent with the default behavior of the xrce-dds client on real vehicle.
+This mismatch can be fixed by manually using `PX4_MICRODDS_NS` on the first instance or by starting adding vehicles from index `1` instead of `0` (this is the default behavior adopted by [sitl_multiple_run.sh](https://github.com/PX4/PX4-Autopilot/blob/main/Tools/simulation/gazebo-classic/sitl_multiple_run.sh) for Gazebo Classic).
+
+The default client configuration in simulation is summarized as follows
+
+| `PX4_MICRODDS_NS` | `px4_instance` | `XRCE_DDS_KEY`   | client namespace      |
```

Comment:
Should this be `px4_instance` as first column?

Reply:
Yes, it could be. I did not think about the order of the two. You proposed that because `px4_instance` is sort of "more important"?

Reply:
No, I proposed it because it seems to me that the instance is what decides the namespace and key.  If you have a single instance setup you don't care about the other values, so it is what generally determines if the others are relevant. Make sense?

Reply:
If I have a single instance setup I may still want to add a custom namespace with `PX4_MICRODDS_NS`. But indeed `px4_instance` is the only one that affects both `XRCE_DDS_KEY`and the final client namespace.

Reply:
UP to you! I'm going to leave this for you now.

Reply:
I'm going to leave it as it is then.